### PR TITLE
feat: Added relay trust and pii config to org/project settings

### PR DESCRIPTION
**Pull request by mitsuhiko**
This is a featured gated UI for relay trust and pii config.

### CODE REVIEW

File: src/sentry/api/endpoints/project_details.py [hunk 1]
```diff
@@ -353,6 +354,10 @@ def put(self, request, project):
         if result.get('storeCrashReports') is not None:
             if project.update_option('sentry:store_crash_reports', result['storeCrashReports']):
                 changed_proj_settings['sentry:store_crash_reports'] = result['storeCrashReports']
+        if result.get('relayPiiConfig') is not None:
+            if project.update_option('sentry:relay_pii_config', result['relayPiiConfig']):
+                changed_proj_settings['sentry:relay_pii_config'] = result['relayPiiConfig'].strip(
+                ) or None
```
Comment (markstory) [inline src/sentry/api/endpoints/project_details.py]: Should we do any validation on the structure of the JSON blob they are giving us?

**Comment by mitsuhiko** [comment]
Jap. I did not find a way to do that yet though. Keeping this pr as a wip
for now.

**Comment by untitaker** [comment]
@mitsuhiko do we want to validate PII config for now? It seems like we would have to pull in marshal for that.

**Review by untitaker — APPROVED** [review]
It has tests like any other config option now, so for me it's fine

**Comment by mitsuhiko** [comment]
@untitaker yeah. validation would be great but that would require some work on the UI components. Not sure if @billyvg already has some ideas how that could work. I was thinking ideally we make a JSON input element that runs a basic JSON syntax validation on the client and when that passes runs a server side validation function on timeout.

**Comment by untitaker** [comment]
What do you want to validate besides basic JSON syntax? We can't validate if a rule name is valid, because relay might run a newer version of marshal which might have more rules built-in.

**Comment by mitsuhiko** [comment]
For instance if regexes are malformed etc. There are quite a few errors that can be created by an invalid config.

**Comment by untitaker** [comment]
@markstory there is no JSON schema for this, and it would be a thing we would have to maintain separately from our current "serde schema", e.g. the type declarations here: https://github.com/getsentry/marshal/blob/master/src/protocol/types.rs#L771

(at some point we'll have to have C bindings for the entire thing anyway)

**Comment by jan-auer** [comment]
@untitaker @markstory I'd say this is one of those things we can easily add once we migrate marshal / PII / meta to a custom parsing implementation away from serde. I think this has to happen reasonably soon. Until then, we assume that people only input valid data. 

However, it would be very straight forward to expose a python function that the config as a whole, but it won't give as good error output as with jsonschema (i.e. it will stop for the first error).

**Comment by jan-auer** [comment]
I'd be fine with just merging for now since it's feature flagged and we will provide guidance for first users. Any kind of highlighting and and validation could then be added in follow-up PRs, once we've decided on the right approach.

**Review by jan-auer — APPROVED** [review]
Can't log on to Percy now, if someone could please check the visual diff.

**Review by untitaker — COMMENTED** [review]
Actually, do we want to feature-gate serverside as well? One could start using relays if they just look at this PR and figure out the API call (assuming we do open registration as part of a closed beta)

File: src/sentry/static/sentry/app/data/forms/organizationGeneralSettings.jsx [hunk 2]
```diff
@@ -210,6 +210,21 @@ const formGroups = [
         ),
         visible: ({features}) => features.has('event-attachments'),
       },
+      {
+        name: 'trustedRelays',
+        type: 'string',
+        multiline: true,
+        autosize: true,
+        maxRows: 10,
+        placeholder: t('paste the relay public keys here'),
```
Comment (billyvg) [inline src/sentry/static/sentry/app/data/forms/organizationGeneralSettings.jsx]: Capitalize?

**Review by billyvg — COMMENTED** [review]
@mitsuhiko It should be pretty easy to add simple validation to the input field (e.g. just attempt to call `JSON.parse).

**Pull request merged**